token/jwt/jwt.go

```diff
@@ -61,12 +61,26 @@ func (j *RS256JWTStrategy) Generate(ctx context.Context, claims MapClaims, heade
 
 // Validate validates a token and returns its signature or an error if the token is not valid.
 func (j *RS256JWTStrategy) Validate(ctx context.Context, token string) (string, error) {
-	return validateToken(token, &j.PrivateKey.PublicKey)
+	switch t := j.PrivateKey.(type) {
+	case *rsa.PrivateKey:
+		return validateToken(token, t.PublicKey)
+	case jose.OpaqueSigner:
+		return validateToken(token, t.Public().Key)
```

Comment:
Is `t.Public()` always set for `jose.OpaqueSigner` or could it cause panics?

Reply:
Yes it is 
https://github.com/square/go-jose/blob/v2/cryptosigner/cryptosigner.go#L35-L51